### PR TITLE
Remove blocking behaviour from the CORS valve

### PR DESCRIPTION
**Pull request by sumedhe**
### Proposed changes in this pull request
Remove blocking behaviour from the CORS valve. After this fix, the cors valve will only add the relevant headers to the response, and the blocking will be happened only in the browser level.

Request types:
- **SIMPLE**: Check the origin, set `Access-Control-Allow-Origin` and `Vary` headers and continue
- **ACTUAL**: Handles similar to the SIMPLE requests.
- **PREFLIGHT**: Set `Access-Control-Allow-Origin`, `Access-Control-Allow-Methods`, `Access-Control-Allow-Headers` and `Access-Control-Max-Age` header and return.
- **NOCORS**: Not setting any cors related headers and continue

**Related Issue:** wso2/product-is#14490

### CODE REVIEW

File: components/org.wso2.carbon.identity.cors.valve/src/main/java/org/wso2/carbon/identity/cors/valve/constant/RequestMethod.java [hunk 1]
```diff
@@ -23,6 +23,21 @@
  */
 public class RequestMethod {
 
+    /**
+     * "GET" method.
+     */
```
Comment (mevan-karu) [inline components/org.wso2.carbon.identity.cors.valve/src/main/java/org/wso2/carbon/identity/cors/valve/constant/RequestMethod.java:28]: Do we need multiline comments for these?

File: components/org.wso2.carbon.identity.cors.valve/src/main/java/org/wso2/carbon/identity/cors/valve/model/CORSRequestType.java [hunk 2]
```diff
@@ -27,5 +24,6 @@
 public enum CORSRequestType {
     ACTUAL,
     PREFLIGHT,
-    OTHER
+    SIMPLE,
+    NON_CORS
```
Comment (mevan-karu) [inline components/org.wso2.carbon.identity.cors.valve/src/main/java/org/wso2/carbon/identity/cors/valve/model/CORSRequestType.java:28]: `NOT_CORS` WDYT?

File: components/org.wso2.carbon.identity.cors.valve/src/main/java/org/wso2/carbon/identity/cors/valve/internal/handler/CORSRequestHandler.java [hunk 3]
```diff
@@ -59,36 +55,20 @@ private CORSManager getCORSManager() {
      * @throws CORSException
      * @throws CORSManagementServiceException
      */
-    public void handleActualRequest(HttpServletRequest request, HttpServletResponse response)
+    public void handleSimpleRequest(HttpServletRequest request, HttpServletResponse response)
```
Comment (mevan-karu) [inline components/org.wso2.carbon.identity.cors.valve/src/main/java/org/wso2/carbon/identity/cors/valve/internal/handler/CORSRequestHandler.java:58]: Why have we changed a public method signature?

**Comment by mevan-karu** [comment]
Don't we need to introduce a config to keep backward compatibility?

File: pom.xml [hunk 4]
```diff
@@ -345,7 +345,7 @@
         <org.wso2.carbon.identity.cors.valve.version>${project.version}</org.wso2.carbon.identity.cors.valve.version>
 
         <!--Carbon identity version-->
-        <identity.framework.version>5.21.63</identity.framework.version>
+        <identity.framework.version>5.23.28</identity.framework.version>
```
Comment (sumedhe) [inline pom.xml:348]: TODO: Bump the version

File: components/org.wso2.carbon.identity.cors.valve/src/main/java/org/wso2/carbon/identity/cors/valve/constant/RequestMethod.java (hunk 1 above)
Comment (sumedhe) [reply components/org.wso2.carbon.identity.cors.valve/src/main/java/org/wso2/carbon/identity/cors/valve/constant/RequestMethod.java]: Removed

File: components/org.wso2.carbon.identity.cors.valve/src/main/java/org/wso2/carbon/identity/cors/valve/model/CORSRequestType.java (hunk 2 above)
Comment (sumedhe) [reply components/org.wso2.carbon.identity.cors.valve/src/main/java/org/wso2/carbon/identity/cors/valve/model/CORSRequestType.java]: +1. Updated.

File: components/org.wso2.carbon.identity.cors.valve/src/main/java/org/wso2/carbon/identity/cors/valve/internal/handler/CORSRequestHandler.java (hunk 3 above)
Comment (sumedhe) [reply components/org.wso2.carbon.identity.cors.valve/src/main/java/org/wso2/carbon/identity/cors/valve/internal/handler/CORSRequestHandler.java]: Reverted.

**Comment by sumedhe** [comment]
> Don't we need to introduce a config to keep backward compatibility?

Discussed and decided not to make this as configurable and remove the blocking behavior as this is an invalid behavior.

**Review by GANGANI — APPROVED** [review]

**Pull request merged**